### PR TITLE
feat(): Adds confirmation screen for 'increaseAllowance'

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
~**NOTE: This PR is blocked by [changes to core repo](https://github.com/MetaMask/core/pull/4069) as well as to the [test dApp](https://github.com/MetaMask/test-dapp/pull/304) first.**~ 

## **Description**

Reuses the token approve confirmation screen for 'increaseAllowance'. It also includes an e2e test for the complete flow.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23560?quickstart=1)

## **Related issues**

Fixes: [#2224](https://github.com/MetaMask/MetaMask-planning/issues/2224)

## **Manual testing steps**

1. Go to the test dApp
2. Create a token
3. Approve token for a small amount
4. Try to transfer a higher amount of tokens from another account. (it should fail)
5. Go back to the first account and increase the allowance.
6. Try to transfer again from the second account. (it should succeed)

### **Before**

Screenshot 2024-03-18 at 17 54 12

### **After**

Screenshot 2024-03-18 at 17 54 38

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]
great work 🚀

**Comment by pedronfigueiredo** [comment]
@metamaskbot update-policies

File: app/scripts/lib/transaction/metrics.ts [hunk 1]
```diff
@@ -848,6 +848,7 @@ async function buildEventFragmentProperties({
       TransactionType.tokenMethodSetApprovalForAll,
       TransactionType.tokenMethodTransfer,
       TransactionType.tokenMethodTransferFrom,
+      TransactionType.tokenMethodIncreaseAllowance,
```
Comment (digiwand) [inline app/scripts/lib/transaction/metrics.ts:851]: [nit] could move up to maintain alphabetical order. same comment to other instances below

File: test/e2e/tests/tokens/increase-token-allowance.spec.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+      WINDOW_TITLES.ExtensionInFullScreenView,
+    );
+    await driver.clickElement({ tag: 'button', text: 'Activity' });
+
+    await driver.waitForSelector({
+      css: '.transaction-list__completed-transactions .activity-list-item [data-testid="activity-list-item-action"]',
+      text: 'Send TST',
+    });
+  }
+});
```
Comment (digiwand) [inline test/e2e/tests/tokens/increase-token-allowance.spec.js:277]: Nice 🙌🏼

**Review by digiwand — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

**Review by segun — DISMISSED** [review]

**Review by seaona — COMMENTED** [review]
Explanation on gas changes required in the e2e: in this PR we are updating the test-dapp to the latest version. In the new version, we removed the suggested gas from the test dapp, to fix a non-related issue with network congestion, meaning that from now on, the gas for the approve and transfer is calculated differently

[https://github.com/MetaMask/test-dapp/commit/a3a2ea834b8739005325901d48f19b551b02a609#diff-bfe9874d239014961b1ae4e8[…]34a410aaaa2ebe3cf89820556L1757](https://github.com/MetaMask/test-dapp/commit/a3a2ea834b8739005325901d48f19b551b02a609#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L1757)

so we need to update both the transfer and approve occurrences where we check the gas, to the new gas values --> those will always be the same (bc ganache setup and mocks are the same)  but they are not the same as the ones we currently have with the previous dapp version

extra note: not sure why FF is passing, but this might indicate that we are not properly validating in that browser - maybe in that browser something else is going on that misses it, but I think that could be out of scope of this PR and could be investigated separately

**Review by jpuri — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**